source/common/network/io_socket_handle_impl.cc

```diff
@@ -335,6 +336,11 @@ Api::IoCallUint64Result IoSocketHandleImpl::recvmsg(Buffer::RawSlice* slices,
     // Get overflow, local address and gso_size from control message.
     for (struct cmsghdr* cmsg = CMSG_FIRSTHDR(&hdr); cmsg != nullptr;
          cmsg = CMSG_NXTHDR(&hdr, cmsg)) {
+      if (cmsg->cmsg_type == save_cmsg_config.optname &&
```

Comment:
Does this "if" return false when there is no config set? If so how? (I that because the default value for these two fields is 0, and no such cmsg is valid?

Reply:
Yes, we're relying on the zero defaults. Do you think we should use `optional` and check for presence?

Reply:
I think so. It *probably* doesn't matter too much, but I think using `optional` would be a bit more explicit.